content/docs/pipelines/pipelines-deploy-alternatives.md

```diff
@@ -0,0 +1,178 @@
+    ```
+
+    Kubeflow pipelines applications take a while (~3 minutes) to start.
+
+1. Get public URL of Pipelines UI and use it to use Kubeflow Pipelines
+    ```
+    kubectl describe configmap inverse-proxy-config -n kubeflow | grep googleusercontent.com 
+    ```
+
+
+## Customization
+Customization can be done through Kustomize [Overlay](https://github.com/kubernetes-sigs/kustomize/blob/master/docs/glossary.md#overlay). 
+
+Note - The instruction below assume you installed kubectl v1.14.0 or later, which has native support of kustomize.
+To get latest kubectl, visit [here](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+
+For the following instructions, first clone [Kubeflow Pipelines repo](https://github.com/kubeflow/pipelines),
+and use it as working directory.
+
+### Deploy on GCP with CloudSQL and GCS
+This is recommended for production environments. See
+[here](https://github.com/kubeflow/pipelines/tree/master/manifests/kustomize/env/gcp) for more details. 
+
+### Change deploy namespace
+To deploy Kubeflow Pipelines in namespace FOO,
+- Edit [dev/kustomization.yaml](https://github.com/kubeflow/pipelines/blob/master/manifests/kustomize/env/dev/kustomization.yaml)
+or [gcp/kustomization.yaml](https://github.com/kubeflow/pipelines/blob/master/manifests/kustomize/env/gcp/kustomization.yaml)
+namespace section to FOO
+- Then run 
+```
+kubectl apply -k manifests/kustomize/env/dev # or other env of your choice
+```
+
+### Disable the public endpoint
+By default, the deployment install an [invert proxy agent](https://github.com/google/inverting-proxy) that exposes a public URL. If you want to skip installing it,
+- Comment out the proxy component in the [kustomization.yaml](https://github.com/kubeflow/pipelines/blob/master/manifests/kustomize/base/kustomization.yaml).
+- Then run 
+```
+kubectl apply -k manifests/kustomize/env/dev # or other env of your choice
+```
+
+The UI is still accessible by port-forwarding
+```
+kubectl port-forward -n kubeflow svc/ml-pipeline-ui 8080:80
+```
+and open http://localhost:8080/
+
+
+
+## Uninstall
+You can uninstall Kubeflow Pipelines by running
+```
+export PIPELINE_VERSION={{% kfp-latest-version %}}
+kubectl delete -k github.com/kubeflow/pipelines//manifests/kustomize/env/dev?ref=$PIPELINE_VERSION
+```
+
+Or if you deployed through kustomize
+```
+kubectl delete -k manifests/kustomize/env/dev
+# or
+kubectl delete -k manifests/kustomize/env/gcp
+```
+
+## Best practice maintaining custom manifests
+
+### Maintain a repo for your manifests
+
+Save the following to a source controlled repo.
+
+File `kustomization.yaml`.
+```
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+# Edit the following to change the deployment to your custom namespace.
+namespace: kubeflow
+
+# You can add other customizations here using kustomize.
+
+# Edit ref in the following link to deploy a different version of KFP.
```

Comment:
It's not clear what "KFP" means, to folks outside the development team. Change "KFP" to "Kubeflow Pipelines", here and below.